cmake/adjust_global_compile_flags.cmake

```diff
@@ -228,25 +228,12 @@ if (MSVC)
   endif()
   if (onnxruntime_target_platform STREQUAL "ARM64")
     set(onnxruntime_target_platform "ARM64")
-    if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.26.0")
```

Comment:
The old code actually doesn't make sense. If the target platform is ARM64, we should always use ASM_MARMASM(Microsoft ARM Assembler) no matter what cmake version is there.